src/backend/marsha/core/api.py

```diff
@@ -497,18 +497,34 @@ def update_live_state(
         if not validate_signature(request.headers.get("X-Marsha-Signature"), msg):
             return Response("Forbidden", status=403)
 
+        # Load the video first to return a 404 if not existing
         video = self.get_object()
```

Comment:
I noticed something weird. You take a "pk" argument but you rely on `self.get_object()`. Either the `pk` argument is useless, or it is used to handle an alternative video and something is wrong!?

Reply:
The `pk` argument is here because this method is an `action` with `detail=True` as explain in the DRF documentation https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing

Reply:
Oh that's weird design :astonished:  
Then I would find it clearer to use the video's `pk` in your query below:
```python
updated_rows = Video.objects.filter(
    ~Q(live_state=serializer.validated_data["state"]),
    pk=video.pk,
).update(live_state=serializer.validated_data["state"])
```